spec/models/employer_spec.rb

```diff
@@ -1,17 +1,27 @@
 require "rails_helper"
 
 describe Employer do
-  let(:name) { "My Company" }
-  let(:account_number) { 1234567 }
-  let(:employer) { FactoryBot.build_stubbed(:employer) }
-
   describe "initialisation" do
-    it "has an account number" do
-      expect(employer.account_number).to eq(account_number)
+    describe "valid case" do
+      it "has an account number" do
+        employer = create(:employer, account_number: 1234567)
+
+        expect(employer.account_number).to eq(1234567)
+      end
+
+      it "has an employer name" do
+        employer = create(:employer, name: "My Company")
+
+        expect(employer.name).to eq("My Company")
+      end
     end
 
-    it "has an employer name" do
-      expect(employer.name).to eq(name)
+    describe "invalid cases" do
+      it "fails when trying to create an employer with no account number" do
+        employer = build(:employer, account_number: nil)
+
+        expect(employer.save).to be(false)
+      end
```

Comment:
Similarly I would test that when invalid attributes are assigned, it is invalid:

```ruby
context "when the account number is not 7 digits long" do
  it "is invalid" do
    employer = build_stubbed(:employer, account_number: 123)
    
    expect(employer).to be_invalid
  end
end
```

I would do the same for the other validations too.

There is the [shoulda-matchers gem](https://github.com/thoughtbot/shoulda-matchers) which provides some 1-liner methods for validations. Sometimes these are helpful (sometimes they're not!).